File: src/beaverdash/web/src/types/project.ts
import { User } from "./auth";
import { Team } from "./team";
import { BoardColumn } from "./task";

/**
 * Project interface mapped from PM.Domain.Entities.Project
 */
export interface Project {
  id: string; // UUID
  teamId: string | null; // UUID
  team?: Team | null;
  name: string;
  description: string | null;
  status: string | null;
  progress: number;
  startDate: string | null; // ISO Date String
  dueDate: string | null; // ISO Date String
  isPublic: boolean;
  shareToken: string | null;
  createdByUserId: string; // UUID
  createdByUser?: User | null;
  createdAt: string; // ISO Date String
  updatedAt: string; // ISO Date String
  boardColumns?: BoardColumn[];
}

/**
 * ProjectMember interface mapped from AIAssistantService project_members
 */
export interface ProjectMember {
  projectId: string; // UUID
  userId: string; // UUID
  user?: User | null;
  status: string | null;
  joinedAt: string; // ISO Date String
}

/**
 * Payload for project synchronization via RabbitMQ
 */
export interface ProjectSyncPayload {
  id: string;
  name: string;
  description: string | null;
  status: string | null;
}

File: src/beaverdash/ProjectManagementService/src/PM.Domain/Entities/Project.cs
using PM.Domain.Common;

namespace PM.Domain.Entities;

public class Project : BaseEntity
{
    public Guid Id { get; set; }
    
    public Guid? TeamId { get; set; }
    public Team? Team { get; set; }

    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public int Progress { get; set; }
    
    public DateTime? StartDate { get; set; }
    public DateTime? DueDate { get; set; }
    
    public bool IsPublic { get; set; }
    public string? ShareToken { get; set; }
    
    public Guid CreatedByUserId { get; set; }
    public User? CreatedByUser { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<BoardColumn> BoardColumns { get; set; } = new List<BoardColumn>();
    public ICollection<Sprint> Sprints { get; set; } = new List<Sprint>();
    public ICollection<ProjectDocument> ProjectDocuments { get; set; } = new List<ProjectDocument>();
}
